include/zephyr/sys/bitarray.h

```diff
@@ -146,6 +146,23 @@ int sys_bitarray_test_and_set_bit(sys_bitarray_t *bitarray, size_t bit, int *pre
  */
 int sys_bitarray_test_and_clear_bit(sys_bitarray_t *bitarray, size_t bit, int *prev_val);
 
+/**
```

Comment:
such changes to common code and APIs should be submitted standalone and should not be mixed with SoC and platform changes, this makes it go fast through the review. Mixing things like that just make this almost impossible to move forrward, i.e. I have now to approve both common code and SoC code I do not understand and I am not qualified to review.

Reply:
Bitarray changes are now in #95416